gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/main/java/io/gravitee/am/gateway/handler/common/flow/impl/FlowManagerImpl.java

```diff
@@ -210,5 +237,72 @@ private Policy createPolicy(Step step) {
             return null;
         }
     }
+
+    private void addExecutionFlow(ExtensionPoint extensionPoint, Flow flow, List<Policy> executionPolicies) {
+        Set<ExecutionFlow> existingFlows = policies.get(extensionPoint);
+        if (existingFlows == null) {
+            existingFlows = new HashSet<>();
+        }
+        ExecutionFlow executionFlow = new ExecutionFlow(flow, executionPolicies);
+        existingFlows.remove(executionFlow);
+        existingFlows.add(executionFlow);
+        policies.put(extensionPoint, existingFlows);
+    }
+
+    private void removeExecutionFlow(ExtensionPoint extensionPoint, String flowId) {
+        Set<ExecutionFlow> existingFlows = policies.get(extensionPoint);
+        if (existingFlows == null || existingFlows.isEmpty()) {
+            return;
+        }
+        existingFlows.removeIf(executionFlow -> flowId.equals(executionFlow.getFlowId()));
```

Comment:
why not a simple remove? ExecutionFlow equals & hasCode are based on the FlowId, result will be the same as far as I can say with a simplest code.